ambari-metrics/ambari-metrics-timelineservice/src/main/java/org/apache/ambari/metrics/core/timeline/aggregators/AggregatorUtils.java

```diff
@@ -50,6 +50,11 @@
       for (Double value : metricValues.values()) {
         // TODO: Some nulls in data - need to investigate null values from host
         if (value != null) {
+          // As the null values does not participate in the sum of current metric, NaN is handled similarly.
```

Comment:
why to add new `if` condition if all could be combined in to one like: `if (value != null && !value.isNaN()) {`